dipy/reconst/dti.py

```diff
@@ -1378,12 +1394,28 @@ def wls_fit_tensor(design_matrix, data, return_S0_hat=False):
     """
     tol = 1e-6
     data = np.asarray(data)
-    ols_fit = _ols_fit_matrix(design_matrix)
     log_s = np.log(data)
-    w = np.exp(np.einsum('...ij,...j', ols_fit, log_s))
-    fit_result = np.einsum('...ij,...j',
-                           pinv(design_matrix * w[..., None]),
-                           w * log_s)
+
+    # NOTE: the SVD approach from _ols_fit_matrix is slower 
+    if False:  # use _ols_fit_matrix - slow
+        ols_fit = _ols_fit_matrix(design_matrix)  # ols_fit is "hat matrix", y' = ols_fit @ y
+        w = np.exp(np.einsum('...ij,...j', ols_fit, log_s))  # predicted S
+    else:  # use same method as ols_fit_matrix - fast
```

Comment:
There are two functions, `ols_fit_matrix` and `_ols_fit_matrix`. The latter returns the 'hat matrix', which when multiplied by data y-values, returns predicted 'y-values'. The only purpose for `_ols_fit_matrix` according to docs is "Mainly used to calculate WLS weights". Given that np.linalg.pinv uses SVD anyway, and would appear to be slower method of getting to the weights, I would might suggest actually removingn `_ols_fit_matrix` entirely, since it is confusing and would appear to be used nowhere else at all